pkg/database/databasescriptview.go

```diff
@@ -0,0 +1,195 @@
+			outStr := v.processTablesForModifiedApps(tablesWithActions, v.title, appName, dbType)
+			outputFile := filepath.Join(outputDir, appName+SQLExtension)
+			outputStruct := MakeScriptOutput(outputFile, outStr)
+			outputSlice = append(outputSlice, *outputStruct)
+		} else if appNew != nil && appOld == nil {
+			v.stringBuilder.Reset()
+			outStr := v.GenerateDatabaseScriptCreate(appNew.GetTypes(), dbType, appName)
+			outputFile := filepath.Join(outputDir, appName+SQLExtension)
+			outputStruct := MakeScriptOutput(outputFile, outStr)
+			outputSlice = append(outputSlice, *outputStruct)
+		}
+	}
+	return outputSlice
+}
+
+func (v *ScriptView) processTablesForModifiedApps(tableDetails []TableDetails,
+	title, appName, dbType string) string {
+	outputStr := v.generateDatabaseScriptModify(tableDetails, dbType, appName)
+	return outputStr
+}
+
+func findAddedDeletedRetainedTables(
+	tableMapOld map[string]*sysl.Type,
+	tableMapNew map[string]*sysl.Type,
+	tableDepthMapOld, tableDepthMapNew map[int][]string) []TableDetails {
+	tableDepthsListNew := makeSortedListOfTableDepth(tableDepthMapNew)
+	tableWithAction := []TableDetails{}
+	//Add the retained and added tables
+	for _, depth := range tableDepthsListNew {
+		tableNames := tableDepthMapNew[depth]
+		sort.Strings(tableNames)
+		for _, tableName := range tableNames {
+			found := ifTableExisted(tableName, tableMapOld)
+			if found {
+				tableDetails := MakeTableDetails(tableMapNew[tableName],
+					tableMapOld[tableName], Retain, tableName)
+				tableWithAction = append(tableWithAction, *tableDetails)
+			} else {
+				tableDetails := MakeTableDetails(tableMapNew[tableName],
+					nil, Add, tableName)
+				tableWithAction = append(tableWithAction, *tableDetails)
+			}
+		}
+	}
+
+	return tableWithAction
+}
+
+func ifTableExisted(tableName string, tableMap map[string]*sysl.Type) bool {
+	item := tableMap[tableName]
+	return item != nil
+}
+
+func makeSortedListOfTableDepth(tableDepthMap map[int][]string) []int {
+	depthList := []int{}
+	for depth := range tableDepthMap {
+		depthList = append(depthList, depth)
+	}
+	sort.Ints(depthList)
+	return depthList
+}
+
+func (v *ScriptView) generateDatabaseScriptModify(tableDetails []TableDetails,
+	dbType, appName string) string {
+	v.stringBuilder.WriteString(fmt.Sprintf("/*TITLE : %s*/\n", v.title))
+	v.stringBuilder.WriteString(fmt.Sprint(databaseScriptHeader))
+	appHeader := "\n\n/*-----------------------Relation Model : " +
+		appName + "-----------------------------------------------*/\n"
+	v.stringBuilder.WriteString(fmt.Sprint(appHeader))
+
+	visitedAttributes := make(map[string]string)
+	for _, tableDetail := range tableDetails {
+		switch tableDetail.action {
+		case "ADD":
+			v.writeCreateSQLForATable(tableDetail.name, tableDetail.table.GetRelation(), visitedAttributes)
+		case "RETAIN":
+			v.writeModifySQLForATable(tableDetail.name, tableDetail.table.GetRelation(),
+				tableDetail.tableOld.GetRelation(), visitedAttributes)
+		}
```

Comment:
What if none of the cases match?

Reply:
as discussed added a warning for scenario where none of these two conditions satisfy. We should not be throwing an error here.